### ISSUE TITLE
Intense CPU and memory usage on no jobs

### ISSUE THREAD

**Issue by rszabo** [mperham/sidekiq#1175]
Hi

We've setup Sidekiq 2.13.0 along with Discourse on a 64-bit Gentoo server with 4GB RAM and a CPU with 2 cores. Sidekiq  has got 25 threads but as the primary site is offline there is nothing to do, no threads are busy. 

Looking at the system resources, Sidekiq has 25 forks and the primary use about 17-30% CPU and about 5.1% of all memory.

Monitoring Redis reveals the various polls and requests of Sidekiq, all oblivious to me except popping the default queue. Something out of the log:
1378857818.914481 [0 127.0.0.1:39964] "zrangebyscore" "sidekiq:retry" "-inf" "1378857818.9142838" "LIMIT" "0" "1"
1378857818.914671 [0 127.0.0.1:39964] "zrangebyscore" "sidekiq:schedule" "-inf" "1378857818.9142838" "LIMIT" "0" "1"

TL;DR: What exactly is Sidekiq doing when it has no jobs to perform, and how badly will it perform live if it can't even handle an idle state? The Sidekiq Pro-site boasts with big customers, so either they're accepting the bloat or we have misconfigured something related to Sidekiq/Discourse/Ruby/Gems/etc. 

PS: ruby 2.0.0-p247
